stdlib/public/core/StringCreate.swift

```diff
@@ -247,26 +247,57 @@ extension String {
       initializingFrom: input, isASCII: isASCII)
     return storage.asString
   }
+  
+  internal static func _fromUTF16(
+    _ input: UnsafeBufferPointer<UInt16>,
+    repairing: Bool = true
+  ) -> (String, repairsMade: Bool)? {
+    if input.isEmpty { return ("", repairsMade: false) }
+    guard let (utf8Len, isASCII) = unsafe utf8Length(
```

Comment:
Once we're ready to push forward on this again, a test I just ran locally suggests that adding a callback to resize the allocation if needed (guess all-ascii, then resize by 2x, then by 1.5x) is actually quite a bit faster than sizing up front.

We'll see if I can make that work for String, will require a bit of shenanigans probably.

Reply:
Fun option: port String growth-on-append to `realloc` then use that